api/types.proto

```diff
@@ -465,12 +465,31 @@ enum TaskState {
 	// Remove this message when the states are deemed perfect.
 }
 
+// State of Networks attached to a container
+message ContainerNetworkStatus {
```

Comment:
So, this is a bit more complicated than I had remembered. The result of a single CNI add call (corresponding to a single `api.Tasks.Network`) is a [Result struct](https://github.com/containernetworking/cni/blob/master/pkg/types/current/types.go#L135..L142) which contains (among other things) an array of [`Interfaces`](https://github.com/containernetworking/cni/blob/master/pkg/types/current/types.go#L240..L244) (name, sandbox, max) and an array of [IPs](https://github.com/containernetworking/cni/blob/master/pkg/types/current/types.go#L257..L264) (4 vs 6, reference to the Interfaces array, network and gateway). There's also routes and dns in there. There's a (brief) example under the "Result" heading in [the spec](https://github.com/containernetworking/cni/blob/master/SPEC.md).

There isn't a requirement that `len(Interfaces) == 1`, i.e. a single CNI add call might produce multiple interfaces. It's also possible that `len(Interfaces) < len(IPs)`, i.e. that a single interface might be given multiple IP addresses. There's also more information here (gateways, routes etc) than I have exposed so far.

I'm not really sure how best to approach this. I'm leaning towards a more hierarchical representation of the raw CNI (in particular dereferencing the interface array index) and leaving out the other fields I mentioned for now (they can be added later easily enough). So something of the form:
```
message ContainerNetworkStatus {
	// Network is the Network ID
	string network = 1;

	// Error, if any.
	string error = 2;

	message NetworkInterface {
		string name = ??;

		// Each IP is in dotted decimal ("192.0.2.1") or IPv6 ("2001:db8::68") form.
		repeated string ip = ?? [(gogoproto.customname) = "IPs"];
	};

	repeated NetworkInterface interfaces = ??;
}
```

While I was here I did spot that my construction of an interface name (`eth%d`) to pass to the CNI add call was unnecessary (and probably wrong). The spec requires the plugin to use that name if it is given, but if not can choose its own name(s), which seems like the correct approach to use (and is reflected in the state above).